include/sensors/core/tasks/pressure_driver.hpp

```diff
@@ -251,155 +253,218 @@ class MMR920C04 {
+        -> void {
+        static_cast<void>(bit_utils::bytes_to_int(m.read_buffer.cbegin(),
+                                                  m.read_buffer.cend(),
+                                                  temporary_data_store));
 
-    auto handle_response(const i2c::messages::TransactionResponse &tm) {
-        uint32_t raw_data = 0x00;
-        int32_t data = 0x0;
-        const auto *iter = tm.read_buffer.cbegin();
-        // Pressure is always a three-byte value
-        iter = bit_utils::bytes_to_int(iter, tm.read_buffer.cend(), raw_data);
-        data = static_cast<int32_t>(raw_data >> 8);
-        auto pressure_pascals = mmr920C04::Pressure::to_pressure(data);
-        switch (static_cast<mmr920C04::Registers>(tm.id.token)) {
-            case mmr920C04::Registers::PRESSURE_READ:
-                read_pressure(data);
-                if (sync) {
-                    if (std::fabs(pressure_pascals) -
-                            std::fabs(offset_average) >
-                        threshold_pascals) {
-                        hardware.set_sync();
-                        stop_polling = true;
-                    } else {
-                        hardware.reset_sync();
-                    }
-                }
-                if (report) {
-                    send_pressure(tm.message_index);
-                }
-                if (limited_poll) {
-                    handle_baseline_poll(tm.message_index, pressure_pascals);
-                }
-                break;
-            case mmr920C04::Registers::LOW_PASS_PRESSURE_READ:
-                read_pressure_low_pass(data);
-                send_pressure_low_pass(tm.message_index);
-                break;
-            case mmr920C04::Registers::TEMPERATURE_READ:
-                read_temperature(data);
-                if (report) {
-                    send_temperature(tm.message_index);
-                }
-                if (limited_poll) {
-                    auto temperature_celsius =
-                        mmr920C04::Temperature::to_temperature(data);
-                    handle_baseline_poll(tm.message_index, temperature_celsius);
-                }
-                break;
-            case mmr920C04::Registers::STATUS:
-                read_status(data);
-                send_status(tm.message_index);
-                break;
-            case mmr920C04::Registers::RESET:
-            case mmr920C04::Registers::IDLE:
-            case mmr920C04::Registers::MEASURE_MODE_1:
-            case mmr920C04::Registers::MEASURE_MODE_2:
-            case mmr920C04::Registers::MEASURE_MODE_3:
-            case mmr920C04::Registers::MEASURE_MODE_4:
-            case mmr920C04::Registers::MACRAM_WRITE:
-                break;
+        auto reg_id = utils::reg_from_id<mmr920C04::Registers>(m.id.token);
+
+        int32_t signed_data_store =
+            static_cast<int32_t>(temporary_data_store >> 8);
+
+        auto pressure_read =
```

Comment:
same thing here - can we factor this out into a pressure handler and a temperature handler?